presto-docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -23,47 +20,362 @@ connection properties as appropriate for your setup:
+
+If an Oracle table contains a type that's not listed in the above table, an
+error shows:
+
+ - An ``unsupported-type.handling-strategy`` equals ``FAIL``
+   (default configuration), any queries on the table fail.
+ - An ``unsupported-type.handling-strategy`` equals ``IGNORE``
+   (default configuration), Presto no longer accesses the column.
+ - An ``unsupported-type.handling-strategy`` equals ``CONVERT_TO_VARCHAR``
+   (default configuration), a column value converts to an unbounded ``VARCHAR``.
+
+Presto to Oracle type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports creating tables with the following types in an Oracle database.
+The table shows the mappings from Presto to Oracle data types:
+
+.. note::
+   For types not listed in the table below, Presto can't perform the ``CREATE
+   TABLE`` and ``AS SELECT``operations to insert info into existing
+   tables that use ``Oracle to Presto`` type mapping.
+
+.. list-table:: Presto to Oracle Type Mapping
+  :widths: 20, 20, 60
+  :header-rows: 1
+
+  * - Presto type
+    - Oracle database type
+    - Notes
+  * - ``TINYINT``
+    - ``NUMBER(3)``
+    -
+  * - ``SMALLINT``
+    - ``NUMBER(5)``
+    -
+  * - ``INTEGER``
+    - ``NUMBER(10)``
+    -
+  * - ``BIGINT``
+    - ``NUMBER(19)``
+    -
+  * - ``DECIMAL(p, s)``
+    - ``NUMBER(p, s)``
+    -
+  * - ``REAL``
+    - ``BINARY_FLOAT``
+    -
+  * - ``DOUBLE``
+    - ``BINARY_DOUBLE``
+    -
+  * - ``VARCHAR``
+    - ``NCLOB``
+    -
+  * - ``VARCHAR(n)``
+    - ``VARCHAR2(n CHAR)`` or ``NCLOB``
+    - See :ref:`character mapping`
+  * - ``CHAR(n)``
+    - ``CHAR(n CHAR)`` or ``NCLOB``
+    - See :ref:`character mapping`
+  * - ``VARBINARY``
+    - ``BLOB``
+    -
+  * - ``DATE``
+    - ``DATE``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP``
+    - ``TIMESTAMP(3)``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP WITH TIME ZONE``
+    - ``TIMESTAMP(3) WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+.. _number mapping:
+
+Mapping numeric types
+^^^^^^^^^^^^^^^^^^^^^
+
+An Oracle ``NUMBER(p, s)`` maps to Presto's ``DECIMAL(p, s)`` except in these
+conditions:
+
+ - No precision is specified for the column (example: ``NUMBER`` or
+   ``NUMBER(*)``), unless ``oracle.number.default-scale`` is set. 
+ - Scale (``s`` ) is greater than precision. 
+ - Precision (``p`` ) is greater than 38. 
+ - Scale is negative and the difference between ``p`` and ``s`` is greater than
+   38, unless ``oracle.number.rounding-mode`` is set to a different value than
+   ``UNNECESSARY``. 
+
+If a column meets any of the conditions:
+
+ - An ``unsupported-type.handling-strategy`` equals ``FAIL``
+   (default configuration) any queries on the table fail.
+ - An ``unsupported-type.handling-strategy`` equals ``IGNORE``
+   (default configuration), Presto no longer accesses the column.
+ - An ``unsupported-type.handling-strategy`` equals ``CONVERT_TO_VARCHAR``
+   (default configuration), a column value converts to an unbounded ``VARCHAR``.
```

Comment:
Please remove this one.
It is not related for `Presto to Oracle Type Mapping`. I think someone copy pasted this initially incidentally.

Reply:
Removed.